app/models/user.rb

```diff
@@ -4,4 +4,30 @@ class User < ApplicationRecord
   # :confirmable, :lockable, :timeoutable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable, :validatable
+
+
+  def has_membership_application?
+    membership_applications.size > 0 && (membership_applications.select { |app| app.persisted? }).count > 0
+  end
+
+
+  def has_company?
+    has_membership_application? && (membership_applications.select { |app| app.company && app.company.persisted? }).count > 0
+  end
+
+
+  def membership_application
+    has_membership_application? ? membership_applications.last : nil
+  end
+
+
+  def company
+    has_company? ? membership_application.company : nil
+  end
+
+
+  def is_member?
+    admin? || ( has_membership_application? ? membership_application.status == 'Godkänd' : false)
```

Comment:
I would not treat an `admin` user as a member - I know this is a convenience for auth purposes but it could get us into trouble later, since we really have false logic in this method.

For instance, this method could be used in a situation where it confirm a member that then executes logic that actually requires the user to be an actual member - that logic could present problems if it executes for a user who is an admin but not a member.